### PR TITLE
fix(infinite-hits): fix load more when no more pages

### LINKED ISSUE #1971: instantsearch.js / infiniteHits / loadmore button let you load more pages after end
Bug.

When reaching the end of the results, pressing load more button makes the application "crash" (no more results).

The button should be disabled to prevent this kind of behavior.

In Instantsearch.js : 1.11.0

### PR DESCRIPTION
**Pull request by bobylito**
fixes #1971

http://psychotic-finger.surge.sh/?query=sdhc%2010g%20sandisk%20pixto&hits=3&idx=instant_search&p=0&hierarchical%5BhierarchicalCategories.lvl0%5D%5B0%5D=Cameras%20%26%20Camcorders&nR%5Bpopularity%5D%5B%3E%3D%5D%5B0%5D=0&is_v=1

You can load more in this URL but not more than 2 times :)

cc. @julienpa @vvo

### CODE REVIEW

**Comment by bobylito** [comment]
It seems that there are two missing documentation elements in this PR:
 - [ ] the actual reference in the website
 - [ ] showMoreLabel documentation (overall check that all elements are documented)

**Comment by bobylito** [comment]
@julienpa Added the missing documentation. 

screenshot 2017-02-13 15 00 15

File: src/widgets/infinite-hits/infinite-hits.js [hunk 1]
```diff
@@ -13,19 +13,20 @@ const bem = bemHelper('ais-infinite-hits');
...
- * @function hits
+ * @function infiniteHits
  * @param  {string|DOMElement} options.container CSS Selector or DOMElement to insert the widget
  * @param  {number} [options.hitsPerPage=20] The number of hits to display per page [*]
  * @param  {Object} [options.templates] Templates to use for the widget
- * @param  {string|Function} [options.templates.empty=''] Template to use when there are no results.
- * @param  {string|Function} [options.templates.item=''] Template to use for each result. This template will receive an object containing a single record.
+ * @param  {string|Function} [options.templates.empty=""] Template to use when there are no results.
+ * @param  {string|Function} [options.templates.item=""] Template to use for each result. This template will receive an object containing a single record.
+ * @param  {bool} [options.showMoreLabel="Show more results"] label used on the show more button
```
Comment (julienpa) [inline src/widgets/infinite-hits/infinite-hits.js]: Type should be `string` rather than `bool`

**Comment by bobylito** [comment]
Good catch @julienpa :)

**Comment by vvo** [comment]
LGTM

**Pull request merged**